var/spack/repos/builtin/packages/ipopt/package.py

```diff
@@ -100,4 +102,10 @@ def configure_args(self):
                 '--with-metis-lib=%s' % spec['metis'].libs.ld_flags,
                 '--with-metis-incdir=%s' % spec['metis'].prefix.include])
 
+        # The IPOPT configure file states that '--enable-debug' implies
+        # '--disable-shared', but adding '--enable-shared' overrides
+        # '--disable-shared' and builds a shared library with debug symbols
+        if '+debug' in spec:
+            args.append('--enable-debug')
```

Comment:
should the other case also be explicitly handled? I.e.
```
else:
     args.append('--disable-debug')
```

Reply:
I've added the other case. I agree that it's good practice because it is self-documenting, rather than relying on library defaults.